common/inputs.js

```diff
@@ -486,6 +487,18 @@
                 scope.clearTime = function () {
                     scope.model.value.time = null;
                 }
+
+                scope.inputContainer = document.querySelector('.input-container');
+
+                scope.setDropdownWidth = function () {
```

Comment:
For the other case, you're assuming that there will be just one dropdown on the page and therefore you're just using a class. You could make the assumption here and you could. Then you can use a class instead (preferably the same class name to avoid confusion) and remove the `columnIndex` that you've attached to scope just for this change.

Also can you please add some comments to explain why you have to do this and what are the assumptions in this code?

Reply:
The `columnIndex` has to be used in this case because relying on the class may fail. One of the other hidden boolean inputs can be selected and the width will be unchanged.

Reply:
hmm. so that means class only works for the other case because all the dropdowns are the same size, and the first dropdown on the page is always one of the main dropdown inputs (rather than select-all dropdowns). If all the select-all dropdowns have the same size, you could use a class that is different from other types of dropdowns on the page. right?